### PR TITLE
Wip/hughsie/synaptics cape

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/synaptics-cape/fu-synaptics-cape-device.c [hunk 1]
```diff
@@ -173,38 +188,100 @@ fu_synaptics_cape_device_rc_set_error(const FuCapCmd *rsp, GError **error)
...
+			    "firmware header data has been corrupted");
+		break;
+	case FU_SYNAPTICS_CAPE_ERROR_SFU_IMAGE_CORRUPTION:
+		g_set_error(error,
+			    G_IO_ERROR,
+			    G_IO_ERROR_INVALID_DATA,
+			    "firmware payload data has been corrupted");
+		break;
+	case FU_SYNAPTICS_CAPE_ERROR_SFU_ALREADY_ACTIVE:
+		g_set_error(error, G_IO_ERROR, G_IO_ERROR_FAILED, "failed to active new firmward");
```
Comment (d4s) [inline plugins/synaptics-cape/fu-synaptics-cape-device.c:260]: s/firmward/firmware/

**Review by d4s — DISMISSED** [review]
LGTM.
Tested the re-installation with EPOS ADAPT 1x5.

PS rust structures and generator looks interesting

File: plugins/synaptics-cape/fu-synaptics-cape-device.c (hunk 1 above)
Comment (hughsie) [reply plugins/synaptics-cape/fu-synaptics-cape-device.c]: Fixed, thanks.

**Comment by hughsie** [comment]
> PS rust structures and generator looks interesting

It's simplifying a lot of code -- see https://fwupd.github.io/libfwupdplugin/tutorial.html#the-rustgen-helper -- feedback most welcome.

**Comment by hughsie** [comment]
>  Tested the re-installation with EPOS ADAPT 1x5.

Brilliant thanks. Any chance you could contribute the emulation data so we can add it to CI?

**Comment by CNXT-Simon** [comment]
Hi @hughsie, thanks for cleaning the code. The rustgen keep code simple.

**Pull request merged**